lib/internal/modules/cjs/loader.js

```diff
@@ -201,14 +201,44 @@ function readPackage(requestPath) {
   }
 
   try {
-    return packageMainCache[requestPath] = JSON.parse(json).main;
+    const parsed = JSON.parse(json);
+    packageMainCache[requestPath] = parsed.main;
+    if (experimentalExports) {
+      packageExportsCache.set(requestPath, parsed.exports);
```

Comment:
since win32 uses filepaths for CJS we need to normalize paths most likely to deal with the `\\` vs `/` separators.

Reply:
I think `requestPath` should already be normalized since it's also used as the cache key for the `main` cache (or, alternatively, at worst we would read `package.json` twice for the same path).

Reply:
my concern is `exports: {".\\foo.js": ".\\bar.js"}` not being caught by `require("_/foo.js")`

Reply:
At least for keys I don't think we'd want to support backslashes. They aren't file paths but more-or-less literal name postfixes. And if they are "paths" they'd be URL paths. If we'd support normalization, I assume we'd want to normalize things like `{"./foo.js": "./bar%20zapp.js"}` to translate it into the platform file system path equivalent.

Reply:
We can do that if we document it as URL based and then transformed to file paths. This still seems to not support `require('_\\foo.js')` which currently works on win32 though.

Reply:
Added two blockers to the [tracking issue](https://github.com/jkrems/proposal-pkg-exports/issues/36):

- [ ] Plug the hole on windows where `x\\foo.js` can get to `foo.js` even if it isn't exported.
- [ ] Normalize `exports` mappings in CJS so that URL paths turn into valid file paths.

I'll try to turn those into issues so it's easier to actually track them and reference them in the next PR(s).

Reply:
sounds fine to do separately